src/content/docs/guidelines/components/modules.md

```diff
@@ -629,8 +629,55 @@ It is also possible for a new multi-tool container to be built and added to BioC
 
 If the software is not available on Bioconda a `Dockerfile` MUST be provided within the module directory. We will use GitHub Actions to auto-build the containers on the [GitHub Packages registry](https://github.com/features/packages).
 
-### 8 Misc
+### 8 Testing
 
-#### 8.1 General module code formatting
+#### 8.1 All output channels must be tested
+
+All output channels SHOULD be present in the nf-test snapshot file, or at a minimum, it MUST be verified that the files exist.
+
+#### 8.2 Stub tests
+
+A stub test MUST exist for the module.
+
+#### 8.3 Tags
+
+Tags for any dependent modules MUST be specified to ensure changes to upstream modules will re-trigger tests for the current module.
+
+```
+tag "modules"
+tag "modules_nfcore"
+tag "<tool>"
+tag "<tool>/<subtool>" # Only if there is a subtool
+tag "<dependent_tool>/<dependent_subtool>" # only if there is a tool this module depends on
+```
```

Comment:
````suggestion
```groovy
tag "modules"
tag "modules_nfcore"
tag "<tool>"
tag "<tool>/<subtool>" // Only if there is a subtool
tag "<dependent_tool>/<dependent_subtool>" // Only if there is a tool this module depends on
```
````